fdbserver/KeyValueStoreSQLite.actor.cpp

```diff
@@ -1145,6 +1158,149 @@ struct RawCursor {
 		}
 		return Optional<Value>();
 	}
+
+	Void readRangeResultWriter(KeyRangeRef keys, Reference<IReadRangeResultWriter> resultWriter, int rowLimit) {
+		// Read range implementation using a result writer
+		Arena arena;
+		if (rowLimit == 0)
+			return Void();
+		if (db.fragment_values) {
+			if (rowLimit > 0) { // return results in asc order
+				if (resultWriter->shouldTrace()) {
+					TraceEvent("Nim_Case1KVStore");
+				}
+				DefragmentingReader i(*this, arena, true);
+				i.moveTo(keys.begin);
+				Optional<KeyRef> nextKey = i.peek();
+				while (nextKey.present() && nextKey.get() < keys.end) {
+					Optional<KeyValueRef> kv = i.getNext();
+					if (resultWriter->shouldTrace()) {
+						TraceEvent("Nim_KVStore4")
+						    .detail("Key", kv)
+						    .detail("resWriter", resultWriter.isValid())
+						    .detail("trace", resultWriter->shouldTrace());
+					}
+					// TraceEvent("Nim_readRangeKVStore").detail("key", kv.get().key);
+					std::variant<bool, KeyRef> res = resultWriter->operator()(kv);
```

Comment:
Hmm I wonder if you could `std::move(kv)` here and change the `IReadRangeResultWriter` interface by changing the function call operator to accept an rvalue (`Optional<KeyValueRef>&& kv`), avoiding a copy that is made here.

Reply:
@sfc-gh-mpilman any thoughts on this?

Reply:
I don’t think it’ll make a difference. This type is tiny and I don’t believe that we have a move constructor that can do anything clever here